lib/main_common.pm

```diff
@@ -2827,48 +2827,72 @@ sub load_extra_tests_kernel {
-        loadtest "publiccloud/patch_and_reboot",     run_args => $args;
-        if (get_var('PUBLIC_CLOUD_IMG_PROOF_TESTS')) {
-            loadtest("publiccloud/img_proof", run_args => $args);
-        } else {
-            loadtest "publiccloud/ssh_interactive_start", run_args => $args;
-            loadtest "publiccloud/instance_overview" unless get_var('PUBLIC_CLOUD_IMG_PROOF_TESTS');
+sub load_qam_publiccloud_tests {
+    my $args = OpenQA::Test::RunArgs->new();
+
+    loadtest "publiccloud/download_repos";
+    loadtest "publiccloud/ssh_interactive_init", run_args => $args;
+    loadtest "publiccloud/register_system",      run_args => $args;
+    loadtest "publiccloud/transfer_repos",       run_args => $args;
+    loadtest "publiccloud/patch_and_reboot",     run_args => $args;
+    if (get_var('PUBLIC_CLOUD_IMG_PROOF_TESTS')) {
+        loadtest("publiccloud/img_proof", run_args => $args);
+        return;    # Finalize (i.e. "terraform destroy") is handled internally
+    } elsif (get_var('PUBLIC_CLOUD_LTP')) {
+        loadtest('publiccloud/run_ltp', run_args => $args);
+        return;    # Finalize (i.e. "terraform destroy") is handled internally
+    } elsif (get_var('PUBLIC_CLOUD_CHECK_BOOT_TIME')) {
+        loadtest('publiccloud/boottime', run_args => $args);
+        return;    # Finalize (i.e. "terraform destroy") is handled internally
+    } elsif (get_var('PUBLIC_CLOUD_FIO')) {
+        loadtest('publiccloud/storage_perf', run_args => $args);
+        return;    # Finalize (i.e. "terraform destroy") is handled internally
+    } else {
+        loadtest "publiccloud/ssh_interactive_start", run_args => $args;
+        loadtest "publiccloud/instance_overview" unless get_var('PUBLIC_CLOUD_IMG_PROOF_TESTS');
+
+        if (get_var('PUBLIC_CLOUD_CONSOLE_TESTS')) {
             load_extra_tests_prepare();
-            load_extra_tests_docker()  if (get_var('PUBLIC_CLOUD_CONTAINERS'));
-            load_extra_tests_console() if (get_var('PUBLIC_CLOUD_CONSOLE_TESTS'));
-            loadtest "publiccloud/ssh_interactive_end", run_args => $args;
+            load_extra_tests_console();
+        }
+        if (get_var('PUBLIC_CLOUD_CONTAINERS')) {
+            load_extra_tests_prepare();
+            load_extra_tests_console();
         }
     }
-    elsif (get_var('PUBLIC_CLOUD_LTP')) {
-        loadtest 'publiccloud/run_ltp';
-    }
-    elsif (get_var('PUBLIC_CLOUD_SLES4SAP')) {
-        loadtest 'publiccloud/sles4sap';
-    }
-    elsif (get_var('PUBLIC_CLOUD_ACCNET')) {
-        loadtest 'publiccloud/az_accelerated_net';
-    }
-    elsif (get_var('PUBLIC_CLOUD_CHECK_BOOT_TIME')) {
-        loadtest "publiccloud/boottime";
-    }
-    elsif (get_var('PUBLIC_CLOUD_FIO')) {
-        loadtest 'publiccloud/storage_perf';
-    }
-    elsif (get_var('PUBLIC_CLOUD_IMAGE_LOCATION')) {
-        loadtest "publiccloud/upload_image";
+
+    # Finalize
+    loadtest("publiccloud/ssh_interactive_end", run_args => $args);
```

Comment:
I think you can move this into `else` ( just after [this](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/12232/files#diff-051cb1f26a96a4f724250d9182535d868c691e20963ef91a1a7ca109e9538379R2861) line ) and than can drop all `return;` 's

Reply:
I don't think so. This module should be executed after each of those conditioned modules.

Reply:
I think Anton's suggestion might work and it looks nicer. I dare to change it ;-)

Reply:
> I don't think so. This module should be executed after each of those conditioned modules.

after which exactly ? because it will be **not** executed  after this modules : 
```
if (get_var('PUBLIC_CLOUD_IMG_PROOF_TESTS')) {
        loadtest("publiccloud/img_proof", run_args => $args);
        return;    # Finalize (i.e. "terraform destroy") is handled internally
    } elsif (get_var('PUBLIC_CLOUD_LTP')) {
        loadtest('publiccloud/run_ltp', run_args => $args);
        return;    # Finalize (i.e. "terraform destroy") is handled internally
    } elsif (get_var('PUBLIC_CLOUD_CHECK_BOOT_TIME')) {
        loadtest('publiccloud/boottime', run_args => $args);
        return;    # Finalize (i.e. "terraform destroy") is handled internally
    } elsif (get_var('PUBLIC_CLOUD_FIO')) {
        loadtest('publiccloud/storage_perf', run_args => $args);
        return;    # Finalize (i.e. "terraform destroy") is handled internally
}
```

but only after one which are inside `else {` part , I don't know if it is right or wrong - just suggested how to do the same what patch already doing but with less code :)

Reply:
Sorry, my mistake. Those modules are actually based on `publiccloud::basetest` and it ensures that `$self->_cleanup()` is called at the end. All good!